### PR TITLE
ci: notify when pre-release starts or fails

### PR DESCRIPTION
**Pull request by v1v**
### Why

Notify when things are not working before the release has even started

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -328,6 +334,7 @@ pipeline {
             stage('Notify') {
               options { skipDefaultCheckout() }
               steps {
+                setEnvVar('PRE_RELEASE_STAGE', 'false')
```
Comment (v1v) [inline Jenkinsfile:337]: This flag will ensure the top-level post failure won't trigger again another notification.

File: Jenkinsfile [hunk 2]
```diff
@@ -70,6 +70,12 @@ pipeline {
                 env.ONLY_DOCS = isGitRegionMatch(patterns: [ '.*\\.(asciidoc|md)' ], shouldMatchAll: true)
               }
             }
+
+            whenTrue(params.release) {
+              setEnvVar('PRE_RELEASE_STAGE', 'true')
+              notifyStatus(slackStatus: 'warning', subject: "[${env.REPO}] Release build just started",
+                           body: "Please go to (<${env.BUILD_URL}|here>) to see the build status or wait for further notifications.")
```
Comment (v1v) [inline Jenkinsfile:77]: Notify when a release build has been triggered

File: Jenkinsfile [hunk 3]
```diff
@@ -414,6 +421,11 @@ pipeline {
       // bundlesize id was generated previously with the generateReport step in the lint stage.
       notifyBuildResult(prComment: true, newPRComment: [ 'bundlesize': 'bundlesize.md' ])
     }
+    failure {
+      whenTrue(params.release && env.PRE_RELEASE_STAGE == 'true') {
+        notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] Pre release steps failed", body: "(<${env.RUN_DISPLAY_URL}|Open>)")
+      }
```
Comment (v1v) [inline Jenkinsfile:427]: If any pre-release steps failed then notify something happened

**Comment by v1v** [comment]
jenkins run the tests please

File: Jenkinsfile [hunk 4]
```diff
@@ -414,6 +421,11 @@ pipeline {
       // bundlesize id was generated previously with the generateReport step in the lint stage.
       notifyBuildResult(prComment: true, newPRComment: [ 'bundlesize': 'bundlesize.md' ])
     }
+    failure {
+      whenTrue(params.release && env.PRE_RELEASE_STAGE == 'true') {
+        notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] Pre release steps failed", body: "(<${env.RUN_DISPLAY_URL}|Open>)")
```
Comment (cachedout) [inline Jenkinsfile:426]:
```suggestion
        notifyStatus(slackStatus: 'danger', subject: "[${env.REPO}] Pre-release steps failed", body: "(<${env.RUN_DISPLAY_URL}|Open>)")
```

**Review by cachedout — APPROVED** [review]

**Review by jahtalab — APPROVED** [review]

**Comment by vigneshshanmugam** [comment]
jenkins run the tests please

**Review by vigneshshanmugam — APPROVED** [review]

**Pull request merged**